api/v1beta1/connector_types.go

```diff
@@ -58,11 +58,27 @@ type ConnectorSpec struct {
 	// HServerEndpoint is the endpoint of the HStreamDB server.
 	// For example: "hstreamdb-hserver:6570"
 	// +kube:validation:Required
-	HServerEndpoint string `json:"hserverEndpoint,omitempty"`
+	HServerEndpoint string `json:"hserverEndpoint"`
 
+	// Deprecated: use `Container.Ports` instead.
 	// ContainerPorts is used to specify the ports that will be exposed by the connector container.
 	// +optional
 	ContainerPorts []corev1.ContainerPort `json:"containerPorts,omitempty"`
+
+	// Container is used to specify the connector container configuration.
+	// +optional
+	Container *ConnectorContainer `json:"container,omitempty"`
+}
+
+// ConnectorContainer defines the connector container configuration.
+type ConnectorContainer struct {
+	// Ports is used to specify the ports that will be exposed by the connector container.
+	// +optional
+	Ports []corev1.ContainerPort `json:"ports,omitempty"`
+
+	// Resources is used to specify the resource requirements of the connector container.
+	// +optional
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
```

Comment:
Why not use `type ConnectorContainer *corev1.Container` ? we can set some default value.

Reply:
I was thinking about the fact that the generated CRD definition will have some extra properties that we don't currently use, as we don't want users to specify the container name or args, etc. For now.

Reply:
This type name is `connectorContainer`, but can only set `ports` and `resources`, feels a bit strange.

Reply:
Indeed. So define them directly in `ConnectorSpec` as `Ports` and `Resources`?

Reply:
We already have ConnectorSpec.ContainerPorts, is it necessary to add another ConnectorSpec.Ports ?

Reply:
I deprecated `ContainerPorts` as I don't want to define `ContainerResources`. They're a little long-winded...Using `Ports` and `Resources` may be even better.

Reply:
After discussion, we decided to stick with corev1.Container.